### PR TITLE
Introduce --failed filter for unit tests.

### PR DESCRIPTION
**Pull request by mpoeter**
### Scope & Purpose

This allows to re-run only those tests that failed in the previous test run. The information which tests previously failed is taken from the "UNITTEST_RESULT.json" (if available).

This should work for all jsunity and mocha tests.

- [x] :pizza: New feature for maintainers

### Testing & Verification

- [x] The behavior in this PR was *manually tested*

Link to Jenkins PR run:
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13537/

### CODE REVIEW

File: js/client/modules/@arangodb/testutils/test-utils.js [hunk 1]
```diff
@@ -687,28 +691,34 @@ function loadClusterTestStabilityInfo(results, instanceInfo){
   }
 }
 
+function getTestCode(file, options, instanceInfo) {
```
Comment (dothebart) [inline js/client/modules/@arangodb/testutils/test-utils.js:694]: DRY - nice!

File: js/client/modules/@arangodb/testutils/testing.js [hunk 2]
```diff
@@ -553,6 +564,27 @@ function iterateTests(cases, options) {
   return results;
 }
 
+function getFailedTestCases(options) {
```
Comment (dothebart) [inline js/client/modules/@arangodb/testutils/testing.js:567]: please move this function to `js/client/modules/\@arangodb/testutils/result-processing.js`

File: js/client/modules/@arangodb/testutils/testing.js [hunk 3]
```diff
@@ -553,6 +564,27 @@ function iterateTests(cases, options) {
...
+      return _.pickBy(_.mapValues(obj, v => {
+          if (typeof v === 'object' && v.status === false) {
+            return filter(v);
+          }
+          return undefined;
+        }),
+        v => v !== undefined);
+    };
+    return filter(lastRun);
+  } catch (err) {
```
Comment (dothebart) [inline js/client/modules/@arangodb/testutils/testing.js:582]: if its broken, don't start runnig the tests, abort!

**Review by dothebart — APPROVED** [review]
please fix the function location.

**Review by dothebart — APPROVED** [review]

**Comment by mpoeter** [comment]
Retry: http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13552/

**Comment by mpoeter** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13557/

**Comment by mpoeter** [comment]
Tests blue

**Pull request merged**